app/views/home/index.html.erb

```diff
@@ -1,15 +1,37 @@
+
+<%= render 'layouts/loginerror'%>
+
+<div class="vcenter">
+
 <% if teacher_signed_in? %>
   <h1>welcome <%=current_teacher.email%></h1>
   <%= link_to "Logout",  destroy_admin_session_path, :method => :delete   %>
```

Comment:
if teacher_signed_in?
 destroy_admin_session_path it should be   teacher signout path.